dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/BroadcastClusterInvoker.java

```diff
@@ -47,28 +48,53 @@ public Result doInvoke(final Invocation invocation, List<Invoker<T>> invokers, L
         RpcContext.getContext().setInvokers((List) invokers);
         RpcException exception = null;
         Result result = null;
+        URL url = getUrl();
+        // The value range of broadcast.fail.threshold must be 0～100.
+        // 100 means that an exception will be thrown last, and 0 means that as long as an exception occurs, it will be thrown.
+        int broadcastFailPercent = url.getParameter("broadcast.fail.percent", 100);
+
+        int failThresholdIndex = invokers.size() * broadcastFailPercent / 100;
+        int failIndex = 0;
         for (Invoker<T> invoker : invokers) {
             try {
                 result = invoker.invoke(invocation);
                 if(null != result && result.hasException()){
                     Throwable resultException = result.getException();
                     if(null != resultException){
-                        logger.warn(resultException.getMessage(), resultException);
-                        throw resultException;
+                        exception = getRpcException(result.getException());
+                        if(failIndex == failThresholdIndex){
+                            break;
+                        }else{
+                            logger.warn(exception.getMessage(), exception);
+                            failIndex++;
+                        }
                     }
                 }
-            } catch (RpcException e) {
-                exception = e;
-                logger.warn(e.getMessage(), e);
             } catch (Throwable e) {
-                exception = new RpcException(e.getMessage(), e);
-                logger.warn(e.getMessage(), e);
+                exception = getRpcException(e);
+                if(failIndex == failThresholdIndex){
+                    break;
+                }else{
+                    logger.warn(exception.getMessage(), exception);
+                    failIndex++;
+                }
             }
         }
-        if (exception != null) {
+
+        if (exception != null && failIndex == failThresholdIndex) {

```

Comment:
Should we report the exception in the end even when failIndex does not reach the threshold?

Reply:
Perhaps the threshold is just to control whether other nodes need to be called, and in the end, as long as an exception occurs, it will be thrown. This way of handling is better, do you think?

Reply:
Yes, I think it's more reasonable.

Reply:
Okay, I will modify it.

Reply:
Please cr again, thanks.